packages/garbo/src/tasks/cockroachPrep.ts

```diff
@@ -16,35 +15,91 @@ import {
+  Effect,
   mallPrice,
   myBuffedstat,
+  myEffects,
+  numericModifier,
+  retrieveItem,
   runChoice,
+  toEffect,
+  use,
   visitUrl,
 } from "kolmafia";
 import { garboValue } from "../garboValue";
 import { freeFightFamiliar } from "../familiar";
-import { freeFightOutfit } from "../outfit";
+import { freeFightOutfit, meatTargetOutfit } from "../outfit";
 import { GarboStrategy, Macro } from "../combat";
 import { acquire } from "../acquire";
 
 // Just checking for the gummi effects for now, maybe can check other stuff later?
 function checkAndFixOvercapStats(): void {
-  if (myBuffedstat($stat`Muscle`) >= 100) {
-    if (have($effect`Gummiheart`)) uneffect($effect`Gummiheart`);
-  }
-  if (myBuffedstat($stat`Mysticality`) >= 100) {
-    if (have($effect`Gummibrain`)) uneffect($effect`Gummibrain`);
-  }
-  if (myBuffedstat($stat`Moxie`) >= 100) {
-    if (have($effect`Gummiskin`)) uneffect($effect`Gummiskin`);
-  }
-  if (
-    myBuffedstat($stat`Moxie`) >= 100 ||
-    myBuffedstat($stat`Mysticality`) >= 100 ||
-    myBuffedstat($stat`Muscle`) >= 100
-  ) {
-    uneffect($effect`Having a Ball!`);
-  }
+  const stats = [$stat`Muscle`, $stat`Moxie`, $stat`Mysticality`];
+  const effects: Effect[] = Object.keys(myEffects()).map((effectName) =>
+    toEffect(effectName),
+  );
+
+  stats.forEach((stat) => {
+    while (myBuffedstat(stat) > 100) {
+      if (
+        !have($effect`Mush-Mouth`) &&
+        mallPrice($item`Fun-Guy spore`) < 5_000
+      ) {
+        retrieveItem($item`Fun-Guy spore`);
+        use($item`Fun-Guy spore`);
+      }
+      if (stat === $stat`muscle`) {
+        if (
+          !have($item`decorative fountain`) &&
+          !have($effect`Sleepy`) &&
+          mallPrice($item`decorative fountain`) < 2_000
+        ) {
+          retrieveItem($item`decorative fountain`);
+        }
+        if (!have($effect`Sleepy`)) {
+          use($item`decorative fountain`);
+        }
+      }
+
+      if (stat === $stat`moxie`) {
+        if (
+          !have($item`patchouli incense stick`) &&
+          !have($effect`Far Out`) &&
+          mallPrice($item`patchouli incense stick`) < 2_000
+        ) {
+          retrieveItem($item`patchouli incense stick`);
+        }
+        use($item`patchouli incense stick`);
+
+        if (have($effect`Endless Drool`) && stat === $stat`Moxie`) {
+          uneffect($effect`Endless Drool`);
+        }
+      }
+
+      if (mallPrice($item`Mr. Mediocrebar`) < 2_000 && !have($effect`Apathy`)) {
+        retrieveItem($item`Mr. Mediocrebar`);
+        use($item`Mr. Mediocrebar`);
+      }
+
+      if (have($effect`Feeling Excited`)) uneffect($effect`Feeling Excited`);
+      // Get effect names from myEffects and convert them to Effect instances
+      effects.forEach((ef) => {
+        // Check if the effect modifier includes the stat and not "meat"
+        if (
+          numericModifier(ef, `${stat.toString}`) &&
+          !(
+            numericModifier(ef, "meat drop") > 0 ||
```

Comment:
use `getModifier` rather than `numericModifier` here; compare them to 0 rather than checking if they're falsy, because negative famweight effects exist.. And make them an array that you test with `some` or `every`